src/main/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtils.java

```diff
@@ -349,17 +375,30 @@ private CustomerUpdateActionUtils() {
 
         return buildSetStoreUpdateAction(oldStores, newStores)
             .map(Collections::singletonList)
-            .orElseGet(() -> {
-                if (oldStores != null && newStores != null) {
-                    final List<UpdateAction<Customer>> updateActions =
-                        buildRemoveStoreUpdateActions(oldStores, newStores);
+            .orElseGet(() -> prepareStoreActions(oldStores, newStores));
```

Comment:
This part here is a bit weird. You already call `buildSetStoreUpdateAction` which should return you correct update actions to have a synced stores. And if this update action is empty, then I guess `prepareStoreActions` would also return empty as it generates `addStore` or `removeStore`.

Reply:
The trick in the `buildSetStoreUpdateAction` private method, it's creating only `setStore` action when old customer does not have stores, acting like adding stores. The function flows a little bit weird but results are expected.